nvflare/private/fed/app/simulator/simulator_worker.py

```diff
@@ -79,13 +82,27 @@ def do_one_task(self, client):
                 client_runner = fl_ctx.get_prop(FLContextKey.RUNNER)
                 self.fire_event(EventType.SWAP_IN, fl_ctx)
 
-                interval, task_processed = client_runner.fetch_and_run_one_task(fl_ctx)
-                self.logger.info(f"Finished one task run for client: {client.client_name}")
-
-                # if any client got the END_RUN event, stop the simulator run.
-                if client_runner.end_run_fired or client_runner.asked_to_stop:
-                    stop_run = True
-                    self.logger.info("End the Simulator run.")
+                run_task_tries = 0
+                while True:
+                    interval, task_processed = client_runner.fetch_and_run_one_task(fl_ctx)
+                    self.logger.info(
+                        f"Finished one task run for client: {client.client_name} "
+                        f"interval: {interval} task_processed: {task_processed}"
+                    )
+
+                    # if any client got the END_RUN event, stop the simulator run.
+                    if client_runner.end_run_fired or client_runner.asked_to_stop:
+                        stop_run = True
+                        self.logger.info("End the Simulator run.")
+                        break
+                    else:
+                        if task_processed:
+                            break
+                        else:
+                            run_task_tries += 1
+                            if run_task_tries >= FETCH_TASK_RUN_RETRY:
+                                break
+                            time.sleep(0.5)
```

Comment:
just a note, we have all these random time.sleep throughout our code base, we need to enhance them in the future (have a file that has all these values) so people at least know where to find/modify these.

Reply:
Will address this in future.